### PR TITLE
feat(auth): update examples based on changes made to iam-client-lib a…

### PR DESCRIPTION
**Pull request by dwojno**
…nd iam-cache-server

### CODE REVIEW

**Review by manihagh — APPROVED** [review]

File: server/nestjs/src/auth/auth.strategy.ts [hunk 1]
```diff
@@ -1,17 +1,26 @@
...
-      cacheServerUrl: 'https://volta-identitycache.energyweb.org/',
-      acceptedRoles
+      jwtSecret: fs.readFileSync('private.pem'),
+      jwtSignOptions: {
+        algorithm: 'RS256',
+      },
+      rpcUrl: process.env.RPC_URL || 'https://volta-rpc.energyweb.org/',
+      cacheServerUrl:
+        process.env.CACHE_SERVER_URL ||
+        'https://identitycache-dev.energyweb.org/',
```
Comment (jrhender) [inline server/nestjs/src/auth/auth.strategy.ts:20]: Hmm, not sure it makes sense to change the cacheserver url to dev. Could be weird if the front-end and the back-end are pointing to different cache-servers. And I think it makes sense for the examples to be using the "volta" server. I think we just wait until auth is deployed to the Volta cache server and then deploy these example updates.

Comment (dwojno) [reply server/nestjs/src/auth/auth.strategy.ts:20]: 👍

**Review by jrhender — APPROVED** [review]

**Comment by jrhender** [comment]
@dwojno I updated the iam-client-lib and passport-did-auth versions and I'm running into an issue when I try to run the react example in this branch. 
First, I changed the cacheClient to be using `"https://identitycache-dev.energyweb.org/"` in `App.tsx`
Then when running the react example, I get unauthorized messages. Does the react example work for you? It seems to fail on the initialization, before making a request to the example back-end
image

**Comment by dwojno** [comment]
@jrhender I think we should test and prepare this changes after we push all changes from iam-cache-server to prod

**Comment by jrhender** [comment]
@dwojno I made some progress on this. I think the iam-client-lib login to the cache-server is working (though there might still be errors when the access_token expires...)

I think the issue might now be with the cookie that the app-backend is setting. You can see that I get the IdentityToken back correctly from the iam-client-lib initialization but the auth against the `roles` endpoint fails (and continues to fail clicking the log-in button again, hence the multiple errors)
image

I think the issue might be that the cookie isn't being set or isn't sent back to the backend? The `req.cookies` are empty when I debug:
image

Are you seeing this as well if you run the example client and the example backend locally?

**Comment by dwojno** [comment]
@jrhender I've fixed all the issue, now everything should work fine :) let's just wait for release and we're going to push this :)

**Pull request merged**